articles/mvc-mvp-mvvm/index.md

```diff
@@ -0,0 +1,52 @@
+
+### Importance of Architectural Patterns
+
+A typical native android application (applications that are only compatible with Android OS) will run only on mobile devices. Now compared to desktops and laptops, mobile phones are smaller in size. So not all the content of the android app can be viewed on a single small screen. Different parts of the app are equipped in different [fragments](https://developer.android.com/guide/fragments). The app may need certain specific commands which will access these parts when the user wishes to access them. When they are accessed the whole app has to update its interface with the new data. Hence, android architecture patterns are implemented to modify these crucial parts of the code. This article deals with such architectural patterns which can be implemented to improve the overall project design.
+
+There are many benefits for an application by adding design patterns. It makes the whole source code cleaner and easier to maintain and manage. It simplifies complex codes since many functions will be nested in different classes. It also helps in covering all unit tests. This article will briefly discuss the three most popular patterns: MVC, MVP, and MVVM.
+
+### Types of Patterns
+
+- MVC - [Model View Controller](#model-view-controller)
+- MVP - [Model View Presenter](#model-view-presenter)
+- MVVM - [Model View View-Model](#model-view-view-model)
+
+Let us take a look at these three types one by one.
+
+#### Model View Controller
+
+[MVC](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) pattern was first developed in 1970. As the name describes itself, this pattern has three components - Model, View, and Controller. The **Model** holds the main logic and rules for how the data is changed and manipulated. View and Controller both depend on the Model. **View** refers to the UI layer. It allows you to see the data that is stored in the model. The **Controller** acts as a request handler. It handles all the incoming requests from the user and updates the model to show what kind of data the user wants to see. Then the view displays it on the screen. If the view follows the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) then its main part is to display the data from the model for every user event handled by the controller.
+
+Here the developers can work on different parts of the pattern separately. Hence, work can be completed a lot quicker. This pattern can come in handy if multiple views are required for the application. Many of the elements function together and coordinate with other components. However, since they all rely on each other, if one of the components fails, it may pose a problem for the whole application.
+

```

Comment:
Please include a section here where you discuss the advantages and disadvantages of MVC.
You've discussed it briefly in the para above, but it's better if you include it separately.